x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/Querier.java

```diff
@@ -245,7 +237,7 @@ public void onResponse(Page page) {
             // 1a. trigger a next call if there's still data
             if (cursor != Cursor.EMPTY) {
                 // trigger a next call
-                planExecutor.nextPage(cfg, cursor, this);
+                cursor.nextPage(cfg, client, writeableRegistry, this);
```

Comment:
Why this change? The plan executor is the one that increments the metrics; by calling the cursor this is skipped.
Is this what's causing the bug? Wouldn't it be easier to fix `planExecutor.nextPage` to stop incrementing total and only next?
The cursor shouldn't be used directly outside the planExecutor - the point of the latter is to abstract low-level details like `writeableRegistry` which now require handling.

Reply:
@costin @Luegg regarding
> Wouldn't it be easier to fix planExecutor.nextPage to stop incrementing total and only next?

I think `total` should still be incremented. This is counting the `rest` calls, be it paging calls with cursor or regular (no pagination) calls, it's still a REST call. This has been like this by design.
The bug lies in counting the "internal" (in the background) calls that ORDER-BY-an-aggregation-function is performing to gather all the data in memory and do the sorting in memory. We assume paging stats should be for end-user paging calls, which is true, but with `ORDER BY SUM` we are using the same mechanism.

What should be happening though, is that PlanExecutor should have a `nextPage` and an "internal" `nextPage`. Both will do the same set of operation except the internal one will not increase the stats counter.

Reply:
As far as I understand, the metric is supposed to count the number of calls o the SQL APIs, not the internal calls that are implementation dependent. Alternatively, we could also have an internal implementation of `PlanExecutor.nextPage` that does not increment the counters and inject `PlanExecutor` into `Querier`.

Reply:
👍 looks like we had the same idea at the same time :)

Reply:
I missed that the change happens inside `LocalAggregationSorterListener`, makes sense now.